ui/packages/lib/src/hooks/useApi.js

```diff
@@ -129,7 +129,8 @@ export const useApi = (
       return {
         cancel: () => {
           didCancel = true;
-        }
+        },
+        promise: promise,
```

Comment:
Capturing the promise and returning it is the only change here, the rest is auto formatting.